### ISSUE TITLE
AWS::StepFunctions::StateMachine: (JSONata) false-positive `E3601` for `Output` in `Choice/Choices`

### ISSUE THREAD

**Issue by r-heimann** [aws-cloudformation/cfn-lint#4206]
### CloudFormation Lint Version

cfn-lint 1.38.1

### What operating system are you using?

Windows 11

### Describe the bug

cfn-lint gives a false-positive `E3601` for `Output` in a `Choice` state.

### Expected behavior

cfn-lint doesn't give a false-positive.

### Reproduction template

```yaml
AWSTemplateFormatVersion: 2010-09-09
Description: Test

Resources:
  StepFunctionRole:
    Type: AWS::IAM::Role
    Properties:
      AssumeRolePolicyDocument:
        Statement:
          - Effect: Allow
            Principal:
              Service: states.amazonaws.com
            Action: sts:AssumeRole

  StateMachine:
    Type: AWS::StepFunctions::StateMachine
    Properties:
      RoleArn: !GetAtt StepFunctionRole.Arn
      Definition:
        QueryLanguage: JSONata # <- Uses JSONata
        StartAt: Pass
        States:
          Pass:
            Type: Pass
            Next: Choice
          Choice:
            Type: Choice
            Choices:
              - Condition: '{% $count($states.input) = 1 %}'
                Output: # <------------------- Gives the error
                  Test: '{% $states.input %}'
                Next: Success
            Default: Success
          Success:
            Type: Succeed
```
